server/svix-server/src/v1/endpoints/message.rs

```diff
@@ -56,15 +57,48 @@ pub fn validate_channels_msg(
     }
 }
 
-pub fn validate_message_in_payload(
-    payload: &serde_json::Value,
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct RawPayload(pub Box<RawValue>);
+
+impl JsonSchema for RawPayload {
+    fn schema_name() -> String {
+        "RawPayload".to_string()
+    }
+
+    fn json_schema(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        serde_json::Value::json_schema(gen)
+    }
+
+    fn is_referenceable() -> bool {
+        false
+    }
+}
+
+impl Eq for RawPayload {}
+
+impl PartialEq for RawPayload {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.get() == other.0.get()
+    }
+}
+
+impl RawPayload {
+    pub fn from_string(val: String) -> serde_json::Result<Self> {
+        Ok(Self(RawValue::from_string(val)?))
+    }
+}
+
+pub fn validate_raw_payload_is_object(
+    payload: &RawPayload,
 ) -> std::result::Result<(), ValidationError> {
-    match payload {
-        serde_json::Value::Object(_) => Ok(()),
-        _ => Err(validation_error(
+    // Verify it's an object/map
+    if payload.0.get().starts_with('{') {
+        Ok(())
```

Comment:
I'm assuming leading whitespace is trimmed automatically with `RawValue`? I saw there were remarks in the docs about how formatting/whitespace is retained as-is, but unsure if that's only the interior of the value.

I'd expect the surrounding whitespace to be removed just based on how I expect the parser to work, but unsure.

Maybe add a small test case just to check the behavior with leading whitespace?

Reply:
There's no concept of interior whitespace, because this is the token so it has to be the value (no whitespace). I also tested it manually.